dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -316,27 +316,30 @@ bool WindowBlockInputStream::returnIfCancelledOrKilled()
 
 Block WindowBlockInputStream::readImpl()
 {
-    const auto & stream = children.back();
-    while (!action.input_is_finished)
-    {
-        if (returnIfCancelledOrKilled())
-            return {};
+    // first try to get result without reading from children
```

Comment:
Is a `if (returnIfCancelledOrKilled())` check needed here?

Reply:
Why? Usually, `readImpl` does not need be aware of cancel signal.